drivers/api/no_os_eeprom.c

```diff
@@ -64,6 +70,9 @@ int32_t no_os_eeprom_init(struct no_os_eeprom_desc **desc,
 		return ret;
 
 	(*desc)->platform_ops = param->platform_ops;
+	(*desc)->mutex = eeprom_mutex_table[param->device_id];
+
+	no_os_mutex_init((*desc)->mutex);
```

Comment:
Do these inits actually work ?
I think they should be `no_os_mutex_init(&(*desc)->mutex);` (*address of a pointer*)

Reply:
added pointer reference